File: backend/src/bin/check_schema.rs
use sqlx::{postgres::PgPoolOptions, Row};
use dotenvy::dotenv;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    dotenv().ok();
    let database_url = std::env::var("DATABASE_URL")?;
    let pool = PgPoolOptions::new()
        .max_connections(5)
        .connect(&database_url)
        .await?;

    let rows = sqlx::query("SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'subjects'")
        .fetch_all(&pool)
        .await?;

    println!("Table: subjects");
    for row in rows {
        let name: String = row.get("column_name");
        let dtype: String = row.get("data_type");
        println!("- {}: {}", name, dtype);
    }

    Ok(())
}

File: backend/check_schema.js
const { Client } = require('pg');
const client = new Client({
    connectionString: 'postgresql://postgres.eyvpvrfadrgnewxslxzo:Alwardas-Polytechnic%402025@aws-1-ap-south-1.pooler.supabase.com:5432/postgres',
    ssl: {
        rejectUnauthorized: false
    }
});
client.connect()
    .then(async () => {
        const tables = ['faculty_subjects', 'lesson_plan_items', 'lesson_plan_progress', 'lesson_schedule'];
        for (const table of tables) {
            const res = await client.query(`SELECT column_name FROM information_schema.columns WHERE table_name = '${table}'`);
            console.log(`${table} Columns:`, res.rows.map(r => r.column_name));
        }
        client.end();
    })
    .catch(err => {
        console.error(err);
        process.exit(1);
    });
